tests/runtime/filter_parser.c

```diff
@@ -28,10 +28,17 @@ char *get_output(void)
     return val;
 }
 
+void clear_output()
+{
+    pthread_mutex_lock(&result_mutex);
+    output = NULL;
+    pthread_mutex_unlock(&result_mutex);
+}
+
```

Comment:
This is to fix following valgrind error.
It means reuse an output of previous test.

```
Test filter_parser_reserve_data_off...          [2023/04/01 07:20:13] [ info] [fluent bit] version=2.1.0, commit=d847afc067, pid=38777
[2023/04/01 07:20:13] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2023/04/01 07:20:13] [ info] [storage] ver=1.2.0, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2023/04/01 07:20:13] [ info] [cmetrics] version=0.5.9
[2023/04/01 07:20:13] [ info] [ctraces ] version=0.3.0
[2023/04/01 07:20:13] [ info] [input:lib:lib.0] initializing
[2023/04/01 07:20:13] [ info] [input:lib:lib.0] storage_strategy='memory' (memory only)
[2023/04/01 07:20:13] [debug] [lib:lib.0] created event channels: read=30 write=31
[2023/04/01 07:20:13] [debug] [lib:lib.0] created event channels: read=34 write=35
[2023/04/01 07:20:13] [ info] [sp] stream processor started
==38777== Invalid read of size 1
==38777==    at 0x4853A6E: strstr (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==38777==    by 0x148317: flb_test_filter_parser_reserve_data_off (filter_parser.c:224)
==38777==    by 0x145D4E: test_do_run_ (acutest.h:1007)
==38777==    by 0x146179: test_run_ (acutest.h:1178)
==38777==    by 0x1473CC: main (acutest.h:1700)
==38777==  Address 0x54a0790 is 0 bytes inside a block of size 128 free'd
==38777==    at 0x484B27F: free (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==38777==    by 0x147ECA: flb_test_filter_parser_extract_fields (filter_parser.c:150)
==38777==    by 0x145D4E: test_do_run_ (acutest.h:1007)
==38777==    by 0x146179: test_run_ (acutest.h:1178)
==38777==    by 0x1473CC: main (acutest.h:1700)
==38777==  Block was alloc'd at
==38777==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==38777==    by 0x5A0166: flb_malloc (flb_mem.h:80)
==38777==    by 0x5A44DE: out_lib_flush (out_lib.c:177)
==38777==    by 0x16E8A9: output_pre_cb_flush (flb_output.h:528)
==38777==    by 0xAE2C4A: co_init (amd64.c:117)
==38777== 
```